projects/pathfinder-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/ProbationAreaService.kt

```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.digital.hmpps.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.entity.ProbationAreaDistrict
+import uk.gov.justice.digital.hmpps.entity.ProbationAreaRepository
+import uk.gov.justice.digital.hmpps.model.LocalDeliveryUnit
+import uk.gov.justice.digital.hmpps.model.ProbationArea
+import uk.gov.justice.digital.hmpps.model.ProbationAreaContainer
+
+@Service
+class ProbationAreaService(private val probationAreaRepository: ProbationAreaRepository) {
+    fun getProbationAreas(): ProbationAreaContainer {
+        val pas = mutableListOf<ProbationArea>()
+        val pad = probationAreaRepository.probationAreaDistricts()
+
+        // create a set of probation areas
+        val paCodes: Set<Pair<String, String>> = pad.map { Pair(it.pCode, it.pDesc) }.toSet()
```

Comment:
Is this `toSet()` call required to remove duplicates from the cross-join?  If so I guess changing `districts` and `boroughs` to `Set` instead of `List` on the entities would have the same effect - not sure if that would simplify things?

Reply:
The set is for the probation areas so I think this is the only way to do it.